fdbclient/VersionVector.h

```diff
@@ -144,9 +171,386 @@ struct VersionVector {
 	bool operator!=(const VersionVector& vv) const { return maxVersion != vv.maxVersion; }
 	bool operator<(const VersionVector& vv) const { return maxVersion < vv.maxVersion; }
 
-	template <class Ar>
-	void serialize(Ar& ar) {
-		serializer(ar, versions, maxVersion);
+	bool compare(const VersionVector& vv) {
+		if (maxVersion != vv.maxVersion) {
+			return false;
+		}
+
+		if (versions.size() != vv.versions.size()) {
+			return false;
+		}
+
+		auto iterA = versions.begin();
+		auto iterB = vv.versions.begin();
+		for (; iterA != versions.end(); iterA++, iterB++) {
+			if (iterA->first != iterB->first || iterA->second != iterB->second) {
+				return false;
+			}
+		}
+		ASSERT(iterB == vv.versions.end());
```

Comment:
shouldn't this be `return iterB == vv.versions.end()`?

Reply:
We already verified that the sizes of the vectors are same, so this is an indication of something wrong - I don't see how this condition can fail at this point if the vectors are of same size.